plugins/bundle/plugin.go

```diff
@@ -429,3 +479,54 @@ func (p *Plugin) configDelta(newConfig *Config) (map[string]*Source, map[string]
 
 	return newBundles, updatedBundles, deletedBundles
 }
+
+func saveBundleToDisk(path string, name string, b *bundle.Bundle) error {
+
+	bundleDir := filepath.Join(path, name)
+
+	// if a bundle already exists, write the new bundle to a temporary file and if successful, replace
+	// the old bundle with the new one
+	if _, err := os.Stat(filepath.Join(bundleDir, "bundle.tar.gz")); err == nil {
+		err := saveCurrentBundleToDisk(bundleDir, ".bundle.tar.gz.tmp", b)
```

Comment:
We should always write to the temp file and do the rename to avoid a case where OPA has written part of it, stopped, restarted, and tries to read the partial file.

Reply:
Done